### ISSUE TITLE
Custom font (Courier New Regular) is not working in iOS and UWP

### ISSUE THREAD

**Issue by JayaramanAyyanar** [xamarin/Xamarin.Forms#3699]
### Description
When using the Courier New Regular custom in forms application which works fine in android and but not in iOS and UWP. We have tested with the label to reproduce the issue which works for the other custom font but not for Courier New Regular. Actually my goal is to set the DataGrid.Columns RecordFont to set as  Courier New Regular custom font where i am unable to achieve my requirement.
### Steps to Reproduce

1. Run the sample.
2. You could find two label in the view.
3. First label is not applied with the custom font and second is working fine with custom font.

### Expected Behavior
Custom font should work for all the platforms of Xamarin.forms
### Actual Behavior
Custom font is not working in iOS and UWP
### Basic Information

- Version with issue: 2.4.0.280
- Last known good version: 3.1
- IDE: Visual Studio 2017 
- Platform Target Frameworks: 
  - iOS:  11.4 
  - Android: 7.0 
  - UWP:  16299 

### Reproduction Link

[CustomFontFrameworkissue.zip](https://github.com/xamarin/Xamarin.Forms/files/2347929/CustomFontFrameworkissue.zip)

**Comment by pauldipietro** [comment]
I'm uncertain of why the last known good version is reported as 3.1 but the described behavior appears to occur with the latest stable on those two platforms using that font.

**Comment by melucas** [comment]
I've been trying to get this to work as well in Shell and my iOS project. 

In the CustomFontFrameworkissue.zip file's project, the iOS project does not have the font set to Build Action = BundleResource and the Info.plist does not contain the UIAppFont key with a the following values:

```
	<key>UIAppFonts</key>
	<array>
	    <string>Amontillados.ttf</string>
	    <string>Courier_New_Regular.ttf</string>
	</array>
```

Should this be the case, or am I following an old example?

When I added the font as above, I can see that the font is getting added to the UIFont.FamilyNames in the iOS AppDelegate.FinishedLaunching() method.  However, when I try to use the font by setting the FontFamily value on a Label, the text it blank.

If I try adding the following to the Shell.Resources/ResourceDictionary nodes in my AppShell.xaml, the iOS crashes and complains about not finding the resource.

```
            <OnPlatform 
                x:Key="HeadingFont" 
                x:TypeArguments="x:String" 
                iOS="Euphorigenic-Regular" 
                Android="euphorigenic.ttf#Euphorigenic" />
```

Which is the correct way to implement a custom ttf font for use with Shell, iOS and Android apps?

**Comment by melucas** [comment]
Some more digging here.  

If I add the "OnPlatform" key to ContentPage.Resources/ResourceDictionary in a content page xaml, the resource is found and the custom font is displayed.

It does NOT work if the key is added to the Application.Resources/ResourceDictionary or Shell.Resources/ResourceDictionary.

**Comment by warappa** [comment]
@JayaramanAyyanar @melucas @pauldipietro

TL;DR: Usage error, everything works as expected (with 2.4.0.280 and 3.5.0.129452)

How to fix the sample:
1. add font to **info.plist** with value _including folder_: `Resources/Courier_New_Regular.ttf`
2. in the font's property-panel select `BundleResource` as **Build Action**
3. change **platform value** to the **font-name** contained _inside_ the font, **_not_** the current font's _filename_: `Courier New` for iOS, `/Assets/Courier_New_Regular.ttf#Courier New` for UWP 

Fixed sample:
[SfDataGridSample-fixed.zip](https://github.com/xamarin/Xamarin.Forms/files/2848394/SfDataGridSample-fixed.zip)

**Comment by jfversluis** [comment]
Seems to work! If there are any more difficulties please feel free to open another issue.
